plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1389,6 +1390,22 @@ public Optional<ConnectorOutputMetadata> finishInsert(
         return Optional.empty();
     }
 
+    private static ImmutableList<String> getWritePartitionColumnNames(List<String> originalPartitionColumns, List<DeltaLakeColumnHandle> dataColumns)
+    {
+        return originalPartitionColumns.stream()
+                .map(columnName -> {
+                    Optional<DeltaLakeColumnHandle> dataColumn = dataColumns.stream()
+                            .filter(column -> columnName.equalsIgnoreCase(column.getPhysicalName()))
+                            .findFirst();
+                    if (!dataColumn.isPresent()) {
```

Comment:
When does this situation happen?

Is this related to `DROP COLUMN` ?  
Not necessarily related, but is there any test coverage for this situation?

Reply:
> When does this situation happen?

Theoretically this is impossible situation, but due to  `originalPartitionColumns` and `dataColumns` composing from different sources some check should be good to have here.

> Is this related to `DROP COLUMN` ?
> Not necessarily related, but is there any test coverage for this situation?

no, unfortunately i have no idea how to cover by test.

Reply:
If it's an impossible situation, we could simplify like this:
```java
                    DeltaLakeColumnHandle dataColumn = dataColumns.stream()
                            .filter(column -> columnName.equalsIgnoreCase(column.getName()))
                            .collect(onlyElement());
```

Reply:
fixed